src/storage/txn/commands/mod.rs

```diff
@@ -432,3 +469,23 @@ impl Debug for Command {
         self.command_ext().fmt(f)
     }
 }
+
+pub trait ReadCommand<E: Engine>: CommandExt {
```

Comment:
Why parameterize over the Engine, seems we only need `E::Snap` so this should parameterize over Snapshot like `WriteCommand`